doc/specs/#2046 - Unified keybindings and commands, and synthesized action names.md

```diff
@@ -0,0 +1,589 @@
+
+<sup>[1]</sup>: This requirement will be relaxed given **Proposal 2**, below,
+but ignored for the remainder of this section, for illustrative purposes.
+
+#### Example
+
+Consider the following settings:
+
+```json
+"commands": [
+  { "name": "Duplicate Tab", "command": "duplicateTab", "keys": "ctrl+alt+a" },
+  { "command": "nextTab", "keys": "ctrl+alt+b" },
+  {
+    "icon": "...",
+    "name": { "key": "NewTabWithProfileRootCommandName" },
+    "commands": [
+      {
+        "iterateOn": "profiles",
+        "icon": "${profile.icon}",
+        "name": { "key": "NewTabWithProfileCommandName" },
+        "command": { "action": "newTab", "profile": "${profile.name}" }
+      }
+    ]
+  },
+  {
+    "icon": "...",
+    "name": "Connect to ssh...",
+    "commands": [
+      {
+        "keys": "ctrl+alt+c",
+        "icon": "...",
+        "name": "first.com",
+        "command": { "action": "newTab", "commandline": "ssh me@first.com" }
+      },
+      {
+        "keys": "ctrl+alt+d",
+        "icon": "...",
+        "name": "second.com",
+        "command": { "action": "newTab", "commandline": "ssh me@second.com" }
+      }
+    ]
+  }
+  {
+    "keys": "ctrl+alt+e",
+    "icon": "...",
+    "name": { "key": "SplitPaneWithProfileRootCommandName" },
+    "commands": [
+      {
+        "iterateOn": "profiles",
+        "icon": "${profile.icon}",
+        "name": { "key": "SplitPaneWithProfileCommandName" },
+        "commands": [
+          {
+            "keys": "ctrl+alt+f",
+            "icon": "...",
+            "name": { "key": "SplitPaneName" },
+            "command": { "action": "splitPane", "profile": "${profile.name}", "split": "automatic" }
+          },
+          {
+            "icon": "...",
+            "name": { "key": "SplitPaneVerticalName" },
+            "command": { "action": "splitPane", "profile": "${profile.name}", "split": "vertical" }
+          },
+          {
+            "icon": "...",
+            "name": { "key": "SplitPaneHorizontalName" },
+            "command": { "action": "splitPane", "profile": "${profile.name}", "split": "horizontal" }
+          }
+        ]
+      }
+    ]
+  }
+]
+```
+
+> 👉 NOTE: The above could have just as easily been placed in `keybindings`, or
+> split across both `keybindings` and `commands`.
+
+This will generate a tree of commands as follows:
+
+```
+<Command Palette>
+├─ Duplicate tab { ctrl+alt+a }
+├─ New Tab With Profile...
+│  ├─ Profile 1
+│  ├─ Profile 2
+│  └─ Profile 3
+├─ Connect to ssh...
+│  ├─ first.com { ctrl+alt+c }
+│  └─ second.com { ctrl+alt+d }
+└─ New Pane... { ctrl+alt+e }
+   ├─ Profile 1...
+   |  ├─ Split Automatically
+   |  ├─ Split Vertically
+   |  └─ Split Horizontally
+   ├─ Profile 2...
+   |  ├─ Split Automatically
+   |  ├─ Split Vertically
+   |  └─ Split Horizontally
+   └─ Profile 3...
+      ├─ Split Automatically
+      ├─ Split Vertically
+      └─ Split Horizontally
+```
+
+Note also the keybindings in the above example:

```

Comment:
I suppose yea, it would be easier for me to just get the keybinding text in the initial command parsing pass, rather then synthesize the action, then reverse lookup the key bound to that action.